lib/ruby_lsp/document.rb

```diff
@@ -30,9 +30,12 @@ def initialize(source:, version:, uri:, encoding: Constant::PositionEncodingKind
       @uri = T.let(uri, URI::Generic)
       @unparsed_edits = T.let([], T::Array[EditShape])
       @syntax_error = T.let(false, T::Boolean)
-      @tree = T.let(SyntaxTree.parse(@source), T.nilable(SyntaxTree::Node))
```

Comment:
Since we are introducing `@parse_result`, I eliminated `@tree`, to reduce the number of state variables needed.